extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -390,6 +396,13 @@ private void processSuccessfulAuthentication(RoutingContext context,
             maxAge += configContext.oidcConfig.authentication.sessionAgeExtension.getSeconds();
         }
         createCookie(context, configContext, getSessionCookieName(configContext), cookieValue, maxAge);
+        fireEvent(SecurityEvent.Type.OIDC_LOGIN, securityIdentity);
+    }
+
+    private void fireEvent(SecurityEvent.Type eventType, SecurityIdentity securityIdentity) {
+        if (resolver.isSecurityEventObserved()) {
+            Arc.container().beanManager().fireEvent(new SecurityEvent(eventType, securityIdentity));
```

Comment:
You should rather create and cache an `Event` object for perf reasons, i.e. `Event<SecurityEvent> event = Arc.container().beanManager().getEvent().select(SecurityEvent.class)` and then `event.fire(...)`.

Reply:
Or even inject the `Event` object if possible...

Reply:
Hi Martin @mkouba so if it gets injected as an `Instance`, ex, `Instance<Event<SecurityEvent>> securityEvent`, then `securityEvent.isResolvable()` (=> if true then it means there must be at least one Observer available) would make the build step with `ValidationPhaseBuildItem` and the `securityEventObserved` property redundant ?

Reply:
Or if injection this `Event` will only lead to a slightly better cache creation time then maybe `Arc.container().beanManager().getEvent().select(SecurityEvent.class)`  will do ?

Reply:
You should not use `Instance` but `Event<SecurityEvent>` (`Event` is a built-in bean and is always available). And no, `securityEvent.isResolvable()` does not care about observers. `Arc.container().beanManager().getEvent().select(SecurityEvent.class)` is fine.

Reply:
Hi @mkouba OK, thanks, so I've just injected the event and also moved the boolean property to the same bean, and now it all looks very compact, and it is also very optimal performance wise :-), how does it all look to you now ? thanks